velox/type/DecimalUtil.h

```diff
@@ -95,16 +95,13 @@ class DecimalUtil {
     if (rescaledValue < -DecimalUtil::kPowersOfTen[toPrecision] ||
         rescaledValue > DecimalUtil::kPowersOfTen[toPrecision] || isOverflow) {
       VELOX_USER_FAIL(
-          "Cannot cast BIGINT '{}' to DECIMAL({},{})",
+          "Cannot cast {} '{}' to DECIMAL({},{})",
+          CppToType<TInput>::name,
```

Comment:
@mbasmanova What is the new recommended way to get a type name here?

Reply:
@karteekmurthys Maybe use `typeid(TInput).name()` here instead, as `CppToType` is going away

Reply:
But this will end up wrong messaging right? `Cannot cast int8_t to DECIMAL`. Velox types to be printed here. @Yuhta Do you think passing TypeKind as a template parameter is a better approach. We can get the TInput type as well as the type name?

Reply:
Adding kind looks a little redundant to me because physical type and kind shall be 1 to 1 in the future.

@mbasmanova Will we have a `CppToKind` trait in the future?  If yes, I think we can keep `CppToType` here temporarily since it's only for numeric physical types, we can swap it out to `CppToKind` easily in the future.

Reply:
That's a good question. We may have something like this.